### PR TITLE
Add observation_space for partial & full observability

### PR DESCRIPTION
**Pull request by hzyjerry**
Thanks for open-sourcing this great repo!

Per request of #5, I added a couple features to this PR:

1. `MultiAgentObservationSpace` class with `contains()` method similar to OpenAI gym
2. Added test cases to: Checker, Combat, OpenAI, pong, predator, switch environments
3. Make environments compatible with latest gym 0.14.0

### CODE REVIEW

**Comment by hzyjerry** [comment]
It seems that the original traffic junction environment is not complete/tested yet, so I left it out in testings.

File: ma_gym/envs/pong_duel/pong_duel.py [hunk 1]
```diff
@@ -32,6 +33,11 @@ def __init__(self, step_cost=0, reward=1, max_rounds=10):
         self.ball_pos = None
         self.__rounds = None
 
+        # agent pos(2), ball pos (2), balldir (6-onehot)
+        self.obs_low = np.array([0., 0., 0., 0.] + [0.] * len(BALL_DIRECTIONS))
+        self.obs_high = np.array([1., 1., 1., 1.] + [1.] * len(BALL_DIRECTIONS))
+        self.observation_space = MultiAgentObservationSpace([spaces.Box(self.obs_low, self.obs_high)for _ in range(self.n_agents)])
```
Comment (koulanurag) [inline ma_gym/envs/pong_duel/pong_duel.py]: Let's make obs_low and obs_high private attributes.

Comment (hzyjerry) [reply ma_gym/envs/pong_duel/pong_duel.py]: Thanks. Addressed

File: tests/envs/test_switch2.py [hunk 2]
```diff
@@ -68,3 +73,20 @@ def test_reset_after_episode_end(env):
     assert step_i == env._step_count
     assert env._total_episode_reward == ep_reward
     test_reset(env)
+
```
Comment (koulanurag) [inline tests/envs/test_switch2.py]:
```suggestion
@pytest_parametrize_plus('env,target_init_obs_n',
                         [(fixture_ref(env), [[0, 0.17], [0, 0.83]]),
                          (fixture_ref(env_full), [[0, 0.17, 0, 0.83], [0, 0.17, 0, 0.83]])])
def test_episode(env, target_init_obs_n):
    env.reset()
    done = [False for _ in range(env.n_agents)]
    ep_reward = [0 for _ in range(env.n_agents)]
    step_i = 0
    while not all(done):
        step_i += 1
        obs_n, reward_n, done, _ = env.step(env.action_space.sample())
        for i in range(env.n_agents):
            ep_reward[i] += reward_n[i]
        assert env.observation_space.contains(obs_n), 'Observation is out of bound'

    assert step_i == env._step_count
    assert env._total_episode_reward == ep_reward

    # testing reset after episode termination
    test_reset(env, target_init_obs_n)
```

File: ma_gym/envs/checkers/checkers.py [hunk 3]
```diff
@@ -34,6 +35,13 @@ def __init__(self, full_observable=False, step_cost=-0.01, max_steps=100):
         self.full_observable = full_observable
 
         self.action_space = MultiAgentActionSpace([spaces.Discrete(5) for _ in range(self.n_agents)])
+        self._obs_high = np.array([1.0, 1.0] + [max(OBSERVATION_MEANING.keys())] * 9)
```
Comment (koulanurag) [inline ma_gym/envs/checkers/checkers.py]: I see that in pong duel , you made the correct change of private (self.__) whereas over here it's protected (self._).
As such, it wouldn't create much of difference for our functionality. However, let's just be consistent with it in all files.

```suggestion
        self.__obs_high = np.array([1.0, 1.0] + [max(OBSERVATION_MEANING.keys())] * 9)
```
same applies for ```self._obs_low```

File: tests/envs/test_switch2.py [hunk 4]
```diff
@@ -68,3 +73,20 @@ def test_reset_after_episode_end(env):
...
+
+def test_partial_observation_space(env):
```
[the remark this answers is not in the document]
Comment (koulanurag) [reply tests/envs/test_switch2.py]: example added above for convience

**Comment by koulanurag** [comment]
> It seems that the original traffic junction environment is not complete/tested yet, so I left it out in testings.

That's True, It can be left out for now.

File: ma_gym/envs/checkers/checkers.py (hunk 3 above)
Comment (hzyjerry) [reply ma_gym/envs/checkers/checkers.py]: Thanks! Addressed

File: requirements.txt [hunk 5]
```diff
@@ -1,8 +1,9 @@
 cloudpickle==1.2.1
 future==0.17.1
-gym==0.13.0
+gym==0.14.0
 numpy==1.16.4
 Pillow==6.0.0
 pyglet==1.3.2
 scipy==1.3.0
 six==1.12.0
+pytest_cases
```
Comment (koulanurag) [inline requirements.txt]: Version must be specified.

In general, I would just keep the testing packages as separate

**Pull request merged**